### ISSUE TITLE
Optimal solution not found for a small MILP

### ISSUE THREAD

**Issue by vitornesello** [ERGO-Code/HiGHS#1213]
Hello,

I have been using HiGHS to solve problems involving timing decisions linked to binary variables.

In some instances, I get sub-optimal solutions whereas, with other solvers, the optimal solution is found.

The `.lp` file below is one of the small instances for which this problem happens:

```
minimize
obj: -1 x55 + 1 x102 + 1000 x115 + 1000 x116 + 1000 x117 + 1000 x118 + 1000 x119 + 1000 x120 + 1 x1 + 1 x2 + 1 x3 + 1 x4 + 1 x5 + 1 x6
subject to
c1_2: 1 x97 - 1 x115 <= 39600
c2_2: 1 x98 - 1 x116 <= 39600
c3_2: 1 x99 - 1 x117 <= 39600
c4_2: 1 x100 - 1 x118 <= 39600
c5_2: 1 x101 - 1 x119 <= 39600
c6_2: 1 x102 - 1 x120 <= 39600
c7_2: -3600 x7 + 1 x61 <= 0
c8_2: 1 x62 <= 0
c9_2: -3600 x9 + 1 x63 <= 0
c10_2: -28800 x10 + 1 x64 <= 0
c11_2: -3600 x11 + 1 x65 <= 0
c12_2: -33400 x12 + 1 x66 <= 0
c13_2: -3600 x14 + 1 x67 <= 0
c14_2: 1 x68 <= 0
c15_2: -3600 x16 + 1 x69 <= 0
c16_2: -28800 x17 + 1 x70 <= 0
c17_2: -3600 x18 + 1 x71 <= 0
c18_1: -33400 x19 + 1 x72 <= 0
c19_1: -3600 x21 + 1 x73 <= 0
c20_1: 1 x74 <= 0
c21_1: -3600 x23 + 1 x75 <= 0
c22_1: -28800 x24 + 1 x76 <= 0
c23_1: -3600 x25 + 1 x77 <= 0
c24_1: -33400 x26 + 1 x78 <= 0
c25_1: -3600 x28 + 1 x79 <= 0
c26_1: 1 x80 <= 0
c27_1: -3600 x30 + 1 x81 <= 0
c28_1: -28800 x31 + 1 x82 <= 0
c29_1: -3600 x32 + 1 x83 <= 0
c30_1: -33400 x33 + 1 x84 <= 0
c31_1: -3600 x35 + 1 x85 <= 0
c32_1: 1 x86 <= 0
c33_1: -3600 x37 + 1 x87 <= 0
c34_1: -28800 x38 + 1 x88 <= 0
c35_1: -3600 x39 + 1 x89 <= 0
c36_1: -33400 x40 + 1 x90 <= 0
c37_1: -3600 x42 + 1 x91 <= 0
c38_1: 1 x92 <= 0
c39_1: -3600 x44 + 1 x93 <= 0
c40_1: -28800 x45 + 1 x94 <= 0
c41_1: -3600 x46 + 1 x95 <= 0
c42_1: -33400 x47 + 1 x96 <= 0
c43: -1 x7 + 1 x8 <= 0
c44: -1 x8 + 1 x9 <= 0
c45: -1 x9 + 1 x10 <= 0
c46: -1 x10 + 1 x11 <= 0
c47: -1 x11 + 1 x12 <= 0
c48: -1 x12 + 1 x13 <= 0
c49: -1 x14 + 1 x15 <= 0
c50: -1 x15 + 1 x16 <= 0
c51: -1 x16 + 1 x17 <= 0
c52: -1 x17 + 1 x18 <= 0
c53: -1 x18 + 1 x19 <= 0
c54: -1 x19 + 1 x20 <= 0
c55: -1 x21 + 1 x22 <= 0
c56: -1 x22 + 1 x23 <= 0
c57: -1 x23 + 1 x24 <= 0
c58: -1 x24 + 1 x25 <= 0
c59: -1 x25 + 1 x26 <= 0
c60: -1 x26 + 1 x27 <= 0
c61: -1 x28 + 1 x29 <= 0
c62: -1 x29 + 1 x30 <= 0
c63: -1 x30 + 1 x31 <= 0
c64: -1 x31 + 1 x32 <= 0
c65: -1 x32 + 1 x33 <= 0
c66: -1 x33 + 1 x34 <= 0
c67: -1 x35 + 1 x36 <= 0
c68: -1 x36 + 1 x37 <= 0
c69: -1 x37 + 1 x38 <= 0
c70: -1 x38 + 1 x39 <= 0
c71: -1 x39 + 1 x40 <= 0
c72: -1 x40 + 1 x41 <= 0
c73: -1 x42 + 1 x43 <= 0
c74: -1 x43 + 1 x44 <= 0
c75: -1 x44 + 1 x45 <= 0
c76: -1 x45 + 1 x46 <= 0
c77: -1 x46 + 1 x47 <= 0
c78: -1 x47 + 1 x48 <= 0
c79: -7200 x1 + 0.014444444444444444 x61 - 0.0038888888888888888 x63 + 0.002777777777777778 x65 + 1 x109 - 1 x121 + 1 x122 <= -692
c80: -7200 x2 + 0.01 x67 - 0.0038888888888888888 x69 + 0.0025 x71 + 1 x110 - 1 x122 + 1 x123 <= -656
c81: -7200 x3 - 0.016666666666666666 x73 - 0.0033333333333333335 x75 + 0.0033333333333333335 x77 + 1 x111 - 1 x123 + 1 x124 <= -886
c82: -7200 x4 + 0.009444444444444445 x79 - 0.006111111111111111 x81 + 0.005277777777777778 x83 + 1 x112 - 1 x124 + 1 x125 <= -365
c83: -7200 x5 + 0.009444444444444445 x85 + 0.0005555555555555556 x87 - 0.001388888888888889 x89 + 1 x113 - 1 x125 + 1 x126 <= -864
c1_1: -27000 x7 + 1 x49 >= 0
c2_1: -3600 x8 + 1 x61 >= 0
c3_1: 1 x62 >= 0
c4_1: -3600 x10 + 1 x63 >= 0
c5_1: -28800 x11 + 1 x64 >= 0
c6_1: -3600 x12 + 1 x65 >= 0
c7_1: -33400 x13 + 1 x66 >= 0
c8_1: -27000 x14 + 1 x50 >= 0
c9_1: -3600 x15 + 1 x67 >= 0
c10_1: 1 x68 >= 0
c11_1: -3600 x17 + 1 x69 >= 0
c12_1: -28800 x18 + 1 x70 >= 0
c13_1: -3600 x19 + 1 x71 >= 0
c14_1: -33400 x20 + 1 x72 >= 0
c15_1: -27000 x21 + 1 x51 >= 0
c16_1: -3600 x22 + 1 x73 >= 0
c17_1: 1 x74 >= 0
c18: -3600 x24 + 1 x75 >= 0
c19: -28800 x25 + 1 x76 >= 0
c20: -3600 x26 + 1 x77 >= 0
c21: -33400 x27 + 1 x78 >= 0
c22: -27000 x28 + 1 x52 >= 0
c23: -3600 x29 + 1 x79 >= 0
c24: 1 x80 >= 0
c25: -3600 x31 + 1 x81 >= 0
c26: -28800 x32 + 1 x82 >= 0
c27: -3600 x33 + 1 x83 >= 0
c28: -33400 x34 + 1 x84 >= 0
c29: -27000 x35 + 1 x53 >= 0
c30: -3600 x36 + 1 x85 >= 0
c31: 1 x86 >= 0
c32: -3600 x38 + 1 x87 >= 0
c33: -28800 x39 + 1 x88 >= 0
c34: -3600 x40 + 1 x89 >= 0
c35: -33400 x41 + 1 x90 >= 0
c36: -27000 x42 + 1 x54 >= 0
c37: -3600 x43 + 1 x91 >= 0
c38: 1 x92 >= 0
c39: -3600 x45 + 1 x93 >= 0
c40: -28800 x46 + 1 x94 >= 0
c41: -3600 x47 + 1 x95 >= 0
c42: -33400 x48 + 1 x96 >= 0
c1: -1 x55 + 1 x97 - 1 x103 = 0
c2: -1 x56 + 1 x98 - 1 x104 = 0
c3: -1 x57 + 1 x99 - 1 x105 = 0
c4: -1 x58 + 1 x100 - 1 x106 = 0
c5: -1 x59 + 1 x101 - 1 x107 = 0
c6: -1 x60 + 1 x102 - 1 x108 = 0
c7: -1200 x1 + 1 x49 + 1 x61 + 1 x62 + 1 x63 + 1 x64 + 1 x65 + 1 x66 - 1 x97 - 1 x109 = 0
c8: -1200 x2 + 1 x50 + 1 x67 + 1 x68 + 1 x69 + 1 x70 + 1 x71 + 1 x72 - 1 x98 - 1 x110 = 120
c9: -1200 x3 + 1 x51 + 1 x73 + 1 x74 + 1 x75 + 1 x76 + 1 x77 + 1 x78 - 1 x99 - 1 x111 = 180
c10: -1200 x4 + 1 x52 + 1 x79 + 1 x80 + 1 x81 + 1 x82 + 1 x83 + 1 x84 - 1 x100 - 1 x112 = 120
c11: -1200 x5 + 1 x53 + 1 x85 + 1 x86 + 1 x87 + 1 x88 + 1 x89 + 1 x90 - 1 x101 - 1 x113 = 180
c12: -1200 x6 + 1 x54 + 1 x91 + 1 x92 + 1 x93 + 1 x94 + 1 x95 + 1 x96 - 1 x102 - 1 x114 = 0
c13: -1 x49 + 1 x56 - 1.0144444444444445 x61 - 1 x62 - 0.9961111111111111 x63 - 1 x64 - 1.0027777777777778 x65 - 1 x66 = 692
c14: -1 x50 + 1 x57 - 1.01 x67 - 1 x68 - 0.9961111111111111 x69 - 1 x70 - 1.0025 x71 - 1 x72 = 536
c15: -1 x51 + 1 x58 - 0.9833333333333333 x73 - 1 x74 - 0.9966666666666667 x75 - 1 x76 - 1.0033333333333334 x77 - 1 x78 = 706
c16: -1 x52 + 1 x59 - 1.0094444444444444 x79 - 1 x80 - 0.9938888888888889 x81 - 1 x82 - 1.0052777777777777 x83 - 1 x84 = 245
c17: -1 x53 + 1 x60 - 1.0094444444444444 x85 - 1 x86 - 1.0005555555555556 x87 - 1 x88 - 0.9986111111111111 x89 - 1 x90 = 684
Bounds
x49 <= 27000
x50 <= 27000
x51 <= 27000
x52 <= 27000
x53 <= 27000
x54 <= 27000
x49 >= 0
x50 >= 0
x51 >= 0
x52 >= 0
x53 >= 0
x54 >= 0
x55 >= 0
x56 >= 0
x57 >= 0
x58 >= 0
x59 >= 0
x60 >= 0
x61 >= 0
x62 >= 0
x63 >= 0
x64 >= 0
x65 >= 0
x66 >= 0
x67 >= 0
x68 >= 0
x69 >= 0
x70 >= 0
x71 >= 0
x72 >= 0
x73 >= 0
x74 >= 0
x75 >= 0
x76 >= 0
x77 >= 0
x78 >= 0
x79 >= 0
x80 >= 0
x81 >= 0
x82 >= 0
x83 >= 0
x84 >= 0
x85 >= 0
x86 >= 0
x87 >= 0
x88 >= 0
x89 >= 0
x90 >= 0
x91 >= 0
x92 >= 0
x93 >= 0
x94 >= 0
x95 >= 0
x96 >= 0
x97 >= 21600
x98 >= 21600
x99 >= 21600
x100 >= 21600
x101 >= 21600
x102 >= 21600
x103 >= 0
x104 >= 0
x105 >= 0
x106 >= 0
x107 >= 0
x108 >= 0
x109 >= 0
x110 >= 0
x111 >= 0
x112 >= 0
x113 >= 0
x114 >= 0
x115 >= 0
x116 >= 0
x117 >= 0
x118 >= 0
x119 >= 0
x120 >= 0
0 <= x121 <= 7200
0 <= x122 <= 7200
0 <= x123 <= 7200
0 <= x124 <= 7200
0 <= x125 <= 7200
0 <= x126 <= 7200
Binary
x1
x2
x3
x4
x5
x6
x7
x8
x9
x10
x11
x12
x13
x14
x15
x16
x17
x18
x19
x20
x21
x22
x23
x24
x25
x26
x27
x28
x29
x30
x31
x32
x33
x34
x35
x36
x37
x38
x39
x40
x41
x42
x43
x44
x45
x46
x47
x48
End
```

**Comment by jajhall** [comment]
Indeed, very odd. I get different incorrect optimal objective values when running HiGHS with/without presolve. Will investigate

**Comment by vitornesello** [comment]
> Indeed, very odd. I get different incorrect optimal objective values when running HiGHS with/without presolve. Will investigate

I have experienced the same behavior.

I have another piece of information that might help in debugging. If you set the upper bounds of binary variables $x_1, ..., x_6$ to be $0$, the objective value decreases, but it still is not optimal.

**Comment by jajhall** [comment]
Thanks. I'm not going to be able to look at this for a while, as I'm in the middle of the busiest few weeks of the year in my real job

**Comment by jajhall** [comment]
Fixed by #1238 

This will propagate to master and next release in due course
